packages/steiger-plugin-fsd/src/public-api/index.ts

```diff
@@ -15,20 +15,38 @@ const publicApi = {
           // The app layer is the top-level layer, there's no need for public API.
           continue
         }
-        for (const segment of Object.values(getSegments(layer))) {
+        for (const [segmentName, segment] of Object.entries(getSegments(layer))) {
           if (getIndex(segment) === undefined) {
-            diagnostics.push({
-              message: 'This segment is missing a public API.',
-              fixes: [
-                {
-                  type: 'create-file',
-                  path: join(segment.path, `index.js`),
-                  // TODO: Infer better content for this file
-                  content: '',
-                },
-              ],
-              location: { path: segment.path },
-            })
+            if (!['ui', 'lib'].includes(segmentName)) {
+              diagnostics.push({
+                message: 'This segment is missing a public API.',
+                fixes: [
+                  {
+                    type: 'create-file',
+                    path: join(segment.path, `index.js`),
+                    // TODO: Infer better content for this file
+                    content: '',
+                  },
+                ],
+                location: { path: segment.path },
+              })
+            } else if (segment.type === 'folder') {
+              for (const child of segment.children) {
+                if (child.type === 'folder' && getIndex(child) === undefined) {
```

Comment:
I saw the discussion on the issue. You mentioned that it should work either with an index file for the segment or with index files in the top-level folder inside the segment. Did you decide to go just with top-level folders, or am I wrong?

Reply:
Also, I didn't find a check for the shared level. Should it work for all ui/lib segments throughout the codebase?

Reply:
> Did you decide to go just with top-level folders, or am I wrong?

That part of the code is only entered when there is no segment-level index, so both are supported.

In the issue discussion, I also mentioned that it should be one or the other, not both, but I later realized that there is no point in enforcing this

As for checking for the Shared layer, the check is implicit. We only enter that code on unsliced layers (App and Shared) and the App layer case exits early

Reply:
Oh, got it

Reply:
And yes, it looks like enforcing one or the other way is unnecessary, I agree